optimum/utils/__init__.py

```diff
@@ -37,6 +40,22 @@ def is_onnxruntime_available():
     return _onnxruntime_available
 
 
+def is_onnxruntime_strict_available():
+    tokenizers_parallelism = os.environ.get("TOKENIZERS_PARALLELISM", "")
+    os.environ["TOKENIZERS_PARALLELISM"] = "false"
+    result = subprocess.check_output(["pip list | grep 'onnxruntime ' || true"], shell=True).decode("utf-8") != ""
+    os.environ["TOKENIZERS_PARALLELISM"] = tokenizers_parallelism
+    return result
+
+
+def is_onnxruntime_gpu_available():
+    tokenizers_parallelism = os.environ.get("TOKENIZERS_PARALLELISM", "")
+    os.environ["TOKENIZERS_PARALLELISM"] = "false"
+    result = subprocess.check_output(["pip list | grep onnxruntime-gpu || true"], shell=True).decode("utf-8") != ""
+    os.environ["TOKENIZERS_PARALLELISM"] = tokenizers_parallelism
+    return result
```

Comment:
I think those functions are quite heavy for just checking that.

Reply:
Agreed, so far I have not find a way directly in Python to know whether `onnxruntime` is installed through `onnxruntime` or `onnxruntime-gpu` package. Will look more into it.

Reply:
Maybe with something shallower? Like:
```python
>>> import pkg_resources
>>> env = {pkg.key for pkg in pkg_resources.working_set}
>>> "onnxruntime" in env
```
What about also adding the check of `onnxruntime-training` ? I think they are in the same category of problem.

Reply:
Thanks! Unfortunately `"onnxruntime" in env` returns True even if there is only onnxruntime-gpu installed. Let me have a look for onnxruntime-training!